src/Microsoft.AspNetCore.Blazor/Components/BlazorComponent.cs

```diff
@@ -115,6 +116,22 @@ protected void StateHasChanged()
         protected virtual bool ShouldRender()
             => true;
 
+        /// <summary>
+        /// Method invoked each time after the component has been rendered.
+        /// </summary>
+        protected virtual void OnAfterRender()
+        {
+        }
+
+        /// <summary>
+        /// Method invoked each time after the component has been rendered. Note that the component does
+        /// not automatically re-render after the completion of any returned <see cref="Task"/>, because
+        /// that would cause an infinite render loop.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing any asynchronous operation, or <see langword="null"/>.</returns>
+        protected virtual Task OnAfterRenderAsync()
+            => null;
```

Comment:
Shouldn't this be Task.CompletedTask?

Reply:
I agree that would be better. There are other equivalent lifecycle methods whose base implementations also return `null`, so in this PR I'll keep this as-is for consistency, and we can review all of them together.

Reply:
@javiercn Thanks, I just wanted to suggest exactly that. :-)